test/test_piece_picker.cpp

```diff
@@ -765,6 +778,19 @@ TORRENT_TEST(reverse_sequential_download)
 			, i % blocks_per_piece));
 }
 
+TORRENT_TEST(reverse_sequential_range_download)
+{
+	// test reverse sequential range download
+	auto p = setup_picker("7654321", "       ", "", "");
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	auto picked = pick_pieces(p, "*******", 7 * blocks_per_piece, 0, nullptr
+		, piece_picker::sequential | piece_picker::reverse, empty_vector);
+	TEST_CHECK(int(picked.size()) == 5 * blocks_per_piece);
```

Comment:
I think the same question applies here. Why not return 7 pieces when 7 are requested?